content/geometry/Point.h

```diff
@@ -34,4 +34,7 @@ struct Point {
 	// returns point rotated 'a' radians ccw around the origin
 	P rotate(double a) const {
 		return P(x*cos(a)-y*sin(a),x*sin(a)+y*cos(a)); }
+	friend ostream &operator<<(ostream &os, P p) {
+		return os << "(" << p.x << "," << p.y << ")";
+	}
```

Comment:
move this `}` up one line for consistency with rotate, I guess, and change `ostream &x` to `ostream& x`